File: 0169/solution.ts
function majorityElement(nums: number[]): number {
  let res = -1;
  let res_count = 0;

  for (let num of nums) {
    if (num != res) {
      if (res_count == 0) {
        res_count++;
        res = num;
      } else {
        res_count--;
      }
    } else {
      res_count++;
    }
  }
  return res;
}

File: 0169/solution.go
package main

func majorityElement(nums []int) int {
	total := -1
	res := -1
	for _, num := range nums {
		if num != res {
			total--
			if total <= 0 {
				total = 1
				res = num
			}
		} else {
			total++
		}
	}
	return res
}